CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/panes/DataAccessPane.java

```diff
@@ -208,9 +209,32 @@ public DataAccessPane(DataAccessViewTopComponent topComponent) {
 
         executeButton.setStyle(GO_STYLE);
         executeButton.setOnAction((ActionEvent event) -> {
+            boolean pluginSelected = false;
+            boolean selectedPluginsValid = true;
+            
+            // check for activated plugins and their validity.
+            for (Tab tab : dataAccessTabPane.getTabs()) {
+                if (tabHasEnabledPlugins(tab)) {
+                    pluginSelected = true;
+                    if (!validateTabEnabledPlugins(tab)) {
+                        selectedPluginsValid = false;
+                    }
+                } 
+            }
+            // when no graph present, create new graph
+            if(graphId == null){
+                if (pluginSelected && selectedPluginsValid) {
+                    NewDefaultSchemaGraphAction graphAction = new NewDefaultSchemaGraphAction();
+                    graphAction.actionPerformed(null);
+                    while(GraphManager.getDefault().getActiveGraph() == null){
```

Comment:
Thanks for looking into this so quickly @aldebaran30701.

Just with this, given we are running this inside setOnAction() something to be mindful of is whether this is running on the JavaFX graphics thread. If so it could lock up the app.

I will do some testing and get back to you but I like the idea.

Reply:
@aldebaran30701 I had a look at this and yes this is running on the FX thread and it does lock up the application. I don't think we can do anything in this case unless we have a progress circle saying that something is happening. I'll say that this is out of scope for this ticket though.

During testing I noticed that when I pressed go, the plugin I had set to run didn't run. I had to click Go again to make the plugin run. Have you seen this? This was happening every single time to me so it didn't work as expected. The test I did was add "www.google.com" to the "Extract Types from Text" plugin. I expected a node on the graph after it created the graph but it's empty.

Can you have a look when you get the chance @aldebaran30701?

Reply:
I've just got around to checking this out. 
I notice what you mean about hanging, it appears frozen but then loads the graph. I am currently unsure how I could go about loading the graph without the delay. If I load it on another thread, the user still waits, only the UI will be usable. 

Great catch! The Graph State was null in the first call to Update(), which then initialized it. I added some initialization code just after the graph ID assigning so it will work on one click. this shouldn't effect load times.

Reply:
Thanks @aldebaran30701 its working great now and very intuitive.

With regards to the delay I tried running some of the work in a different thread and it took longer lol so I think it's current behaviour is something we have to live with.

On web apps now that follow Material UI then have a small loading bar at the top. Something like that would be pretty cool I reckon.